src/index.ts

```diff
@@ -192,13 +184,41 @@ export class ProjectConfig {
    * return a new ProjectConfig instance created with those options.
    */
   static loadConfigFromFile(filepath: string): ProjectConfig|null {
-    let configParsed = ProjectConfig.loadOptionsFromFile(filepath);
+    const configParsed = ProjectConfig.loadOptionsFromFile(filepath);
     if (!configParsed) {
       return null;
     }
     return new ProjectConfig(configParsed);
   }
 
+  /**
+   * Returns the given configJsonObject if it is a valid ProjectOptions object,
+   * otherwise throws an informative error message.
+   */
+  static validateOptions(configJsonObject: {}): ProjectOptions {
+    const validator = new jsonschema.Validator();
+    const result = validator.validate(configJsonObject, getSchema());
+    if (result.errors.length > 0) {
+      throw result.errors[0];
+    }
+    if (result.throwError) {
+      throw new Error(
+          `An error was encountered while validating the ` +
+          `Polymer Project Config.`);
+    }
+    return configJsonObject;
+  }
+
+  /**
+   * Returns a new ProjectConfig from the given JSON object if it's valid.
+   *
+   * Throws if the given JSON object is an invalid ProjectOptions.
+   */
+  static validateAndCreate(configJsonObject: {}) {
```

Comment:
This function name doesn't match with the others:
- `loadOptionsFromFile()` - loads options from file, validates, returns
- `loadConfigFromFile()`- loads options from file, validates, returns a new ProjectConfig instance
-  `validateAndCreate()` - is passed options directly, validates, returns a new ProjectConfig instance

How could we rename without breaking the existing interface? 

Since all validate, maybe just `loadConfig()` for this new one? `createConfig()`?

Reply:
Ah, good point. I named it `validateAndFoo` in contrast to the constructor, but yeah that clashes with the other static methods.

I jumped the gun in releasing :/

Sent over https://github.com/Polymer/polymer-project-config/pull/42 to at least document that we want to change this next major version bump.